### PR TITLE
Add more handlers to IDT in IA32e

### PR DESCRIPTION
**Pull request by alexandrefresnais**
To not go through the pain of rebasing the old branch, and to get a cleaner result, a new branch was created.

### CODE REVIEW

File: src/arch/ia32e/interrupts/frame.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            self.code_segment,
+            self.rflags,
+            self.stack_pointer,
+            self.stack_segment
+        )
+    }
+}
+
+#[repr(C)]
+pub struct InterruptStackFrame {
```
Comment (P-E-P) [inline src/arch/ia32e/interrupts/frame.rs]: I'm not sure why we need this intermediate structure ? Couldn't we simply rename `InterruptStackFrameValue` to `InterruptStackFrame` and remove this structure entirely ?

**Review by P-E-P — COMMENTED** [review]
It looks damn clean, I like that. Appart a point I would like to discuss it's all good.

File: src/arch/ia32e/interrupts/frame.rs (hunk 1 above)
Comment (alexandrefresnais) [reply src/arch/ia32e/interrupts/frame.rs]: Yes, sorry, I should have tried. Fixed in a31d94703ddd20bded81afb35a08691a93636c5b

**Review by P-E-P — APPROVED** [review]
LGTM

**Pull request merged**